ojd_daps_skills/pipeline/skill_ner/prodigy/skill_recipe.py

```diff
@@ -0,0 +1,159 @@
+        for ent in doc.ents:
+            # Ignore if the predicted entity is not in the selected labels.
+            if labels and ent.label_ not in labels:
+                continue
+            # Create a span dict for the predicted entity.
+            spans.append(
+                {
+                    "token_start": ent.start,
+                    "token_end": ent.end - 1,
+                    "start": ent.start_char,
+                    "end": ent.end_char,
+                    "text": ent.text,
+                    "label": ent.label_,
+                }
+            )
+        task["spans"] = spans
+        # Rehash the newly created task so that hashes reflect added data.
+        task = set_hashes(task)
+        yield task
+
+
+# Recipe decorator with argument annotations: (description, argument type,
+# shortcut, type / converter function called on value before it's passed to
+# the function). Descriptions are also shown when typing --help.
+@prodigy.recipe(
+    "ner.correct_skills",
+    dataset=("The dataset to use", "positional", None, str),
+    spacy_model=("The base model", "positional", None, str),
+    source=("The source data as a JSONL file", "positional", None, str),
+    label=("One or more comma-separated labels", "option", "l", split_string),
+    update=("Whether to update the model during annotation", "flag", "UP", bool),
+    exclude=("Names of datasets to exclude", "option", "e", split_string),
+    unsegmented=("Don't split sentences", "flag", "U", bool),
+    component=("Name of NER component in the pipeline", "option", "c", str),
+)
+def ner_correct_skills(
+    dataset: str,
+    spacy_model: str,
+    source: str,
+    label: Optional[List[str]] = None,
+    update: bool = False,
+    exclude: Optional[List[str]] = None,
+    unsegmented: bool = False,
+    component: Optional[str] = "ner",
+):
+    """
+    Create gold-standard data by correcting a model's predictions manually.
+    This recipe used to be called `ner.make-gold`.
+    """
+    # Load the spaCy model.
+    nlp = spacy.load(spacy_model)
+
+    labels = label
+
+    # Get existing model labels, if available.
+    if component not in nlp.pipe_names:
+        raise ValueError(
+            f"Can't find component '{component}' in the provided pipeline."
+        )
+    model_labels = nlp.pipe_labels.get(component, [])
+
+    # Check if we're annotating all labels present in the model or a subset.
+    use_all_model_labels = len(set(labels).intersection(set(model_labels))) == len(
+        model_labels
+    )
+
+    # Load the stream from a JSONL file and return a generator that yields a
+    # dictionary for each example in the data.
+    stream = JSONL(source)
+
+    if not unsegmented:
+        # Custom way to split into chunks of a certain size
+        # its not ideal if these are too big (the model struggles)
+        # or too small (it's hard to label)
+        def split_text(adverts, chunk_size=5):
+            for advert in adverts:
```

Comment:
it's funny how the length is an issue in this recipe but not in the other

Reply:
its so odd!